### ISSUE TITLE
Webpack2: Fetching split code does not use same filename webpack generates

### ISSUE THREAD

**Issue by chandlerprall** [webpack/webpack#1848]
(this is using the ES6 imports with Webpack2)

**Issue: async load of split code requests wrong filenames.** Code splitting generates files: `0`, `1`, `2`, `3` but async load requests `0.0`, `1.1`, `2.2`, `3.3`.

I have a Single Page App where everything is structured in separate node modules. There is one entry point, `client.js`, which needs to detect the appropriate view and load the correct page module. I can structure the modules in such a way that code splitting is not triggered, and in everything works. However, to avoid generating a single file to run the entire site I need to split it into chunks. Easy enough with code splitting, and the relevant portion of `client.js` is

```javascript
function renderView(viewComponentName, urlParams, data) {
    System.import('cartridges/views/' + viewComponentName).then(function(ViewComponent) {
        unmountComponentAtNode($container);
        render(
            createElement( Wrapper, {children: createElement(ViewComponent.component)} ),
            $container
        );
    });
}
```

I have four of these "cartridges" defined currently: page1, page2, page3, and fourohfour. Webpack picks them up just fine and in my output directory I have five files: client.js, 0, 1, 2, and 3. However, when running the app Webpack makes a request for files named 0.0, 1.1, 2.2, and 3.3, generated from the line `script.src = __webpack_require__.p + "" + chunkId + "." + ({}[chunkId]||chunkId) + "";` which is generated from [JsonpMainTemplatePlugin lines 56-74](https://github.com/webpack/webpack/blob/v2.0.2-beta/lib/JsonpMainTemplatePlugin.js#L56).

My webpack.config.js -
```javascript
var path = require('path');
var webpack = require('webpack');

module.exports = {
    context: __dirname,
    entry: {
        'client.js': './client.js',
    },
    output: {
        path: path.join(__dirname, 'bundles'),
        filename: '[name]'
    }
};
```

**Comment by chandlerprall** [comment]
@bebraw sorry if it wasn't clear, this is a bug report around code splitting. I've edited the original post to immediately describe the issue, and then follow with the setup.

**Comment by bebraw** [comment]
@chandlerprall Thanks. Reclassified. :+1:

**Comment by sokra** [comment]
fixed by c838656e7fe20e52f3fee9428b923a25292fd988
